src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -2887,8 +3007,421 @@ public boolean isSlave() {
+						currentCandidate = slaveryEvent;
+					}
+				}
+			}
+		}
+		
+		if(currentCandidate != null)
+		{
+
+			int randomValue = (int )(Math.random() * (GameCharacter.MAX_MINUTES_TO_NEXT_SLAVERY_EVENT - GameCharacter.MIN_MINUTES_TO_NEXT_SLAVERY_EVENT) + 1) + GameCharacter.MIN_MINUTES_TO_NEXT_SLAVERY_EVENT;
+			this.timeOfNextEventCheck = Main.game.getMinutesPassed() + randomValue;
+			
+			return currentCandidate.getTriggerDialogue();
+		}
+		
+		Main.game.setActiveNPC(null);
+		return null;
+	}
+
+	/**
+	 * Returns punishment dialogue node. Doesn't automatically set it like normal; use it for events where player messes up to give them an immediate punishment.
+	 * 
+	 * Supplying reason will allow the punishment events to use it through getPerformanceRemark call.
+	 * @return
+	 */
+	public DialogueNodeOld getPunishmentDialogue(String reason)
+	{
+		this.setOutstandingPerformanceRemark(reason, true);
+		return this.getPunishmentDialogue();
+	}
+	
+	/**
+	 * Adds a rule to the list of rules enforced by this character.
+	 */
+	public void addRule(PlayerSlaveryRule ruleToAdd)
+	{
+		this.slaveOwnerRules.add(ruleToAdd);
+	}
+	
+	/**
+	 * Removes the rule from the list of rules enforced by this character.
+	 */
+	public void removeRule(String ruleUniqueName)
+	{
+		PlayerSlaveryRule ruleToRemove = null;
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getUniqueName() == ruleUniqueName)
+			{
+				ruleToRemove = slaveryRule;
+			}
+		}
+		if(ruleToRemove != null)
+		{
+			this.slaveOwnerRules.remove(ruleToRemove);
+		}
+	}
+	
+	/**
+	 * Removes the rule from the list of rules enforced by this character.
+	 */
+	public void removeRule(PlayerSlaveryRule ruleToRemove)
+	{
+		if(ruleToRemove != null)
+		{
+			this.slaveOwnerRules.remove(ruleToRemove);
+		}
+	}
+	
+	/**
+	 * Checks to see if a slavery rule is enforced by the owner. If it is enforced, returns it; otherwise returns null.
+	 * @param ruleUniqueName
+	 * @return
+	 */
+	public PlayerSlaveryRule hasRule(String ruleUniqueName)
+	{
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getUniqueName() == ruleUniqueName)
+			{
+				return slaveryRule;
+			}
+		}
+		return null;
+	}
+	
+	/**
+	 * Checks to see if a slavery rule is enforced by the owner. If it is enforced, returns it; otherwise returns null.
+	 * @return
+	 */
+	public PlayerSlaveryRule hasRule(PlayerSlaveryRule ruleToFind)
+	{
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getUniqueName() == ruleToFind.getUniqueName())
+			{
+				return slaveryRule;
+			}
+		}
+		return null;
+	}
+	
+	/**
+	 * Returns the full list of rules the owner has.
+	 */
+	public Set<PlayerSlaveryRule> getRules()
+	{
+		return this.slaveOwnerRules;
+	}
+	
+	/**
+	 * Returns true if the character is within the bounds of their owner's property.
+	 */
+	public boolean isWithinOwnersPropery()
+	{
+		if(this.getOwner() == null)
+		{
+			return true;
```

Comment:
why does this return true? add a comment in the method's body

Reply:
Replied above